### DISCUSSION TITLE
Clustermedia cluster list watch multi cluster. Is the memory usage of the informer queue large on the host

### DISCUSSION THREAD

**Discussion by xumujian** [clusterpedia-io/clusterpedia#442] [Q&A]
How much memory does the host need for the cluster running the clustermedia service

**Comment by Iceber** [comment]
clusterpedia replaces the informer's cache store with a store that only stores the resource version of the resource, reducing memory usage.

Of course, resources are cached in memory before they are processed by the storage layer.

**Reply by xumujian** [reply]
How big is the normal cache? If you listen to 100 clusters, each cluster has 8000 pods

**Reply by Iceber** [reply]
The infromer cache usage is about 200 M in total.

The memory usage of the temporary cache before the resources are stored in the storage component depends on the efficiency of the storage layer processing

**Reply by Iceber** [reply]
Another thing to note is that with a size of 100*8000, the informer will also take up a lot of temporary memory when it comes to pulling resources, but these will be reclaimed after a while

**Reply by xumujian** [reply]
Therefore, for centralized multi cluster resource synchronization, the host of clustermpedia needs a lot of memory

**Reply by Iceber** [reply]
For kube pods, consider setting a high limit and a low request, since a normally running clusterpedia doesn't pile up a lot of kube resources and the memory footprint is very low at this time

**Reply by xumujian** [reply]
Our business scenario is mainly to watch the pod resources of multiple clusters, and report the changed pod data in real time when the container status in the pod changes. Now, when there are many clusters, memory may be a bottleneck

**Reply by Iceber** [reply]
Even though clusterpedia has a temporary memory usage during synchronization due to poor performance of the user-provided storage component(mysql/postgres), it still saves memory resources compared to using the native informer directly.

In addition, I'd like to add some concurrent cluster collection limits at startup to avoid a lot of temporary memory usage, but this would result in a longer time for all clusters to fully synchronize, and I'm not sure if this is necessary.

**Reply by xumujian** [reply]
If all events of resource changes in any cluster are written to the database, will database access be too frequent
